evm/vm/forks/sharding/computation.py

```diff
@@ -28,9 +29,24 @@
 
 
 class ShardingComputation(SpuriousDragonComputation):
+    _paygas_gasprice = None
+
     # Override
     opcodes = SHARDING_OPCODES
 
+    def get_PAYGAS_gas_price(self):
+        return self._paygas_gasprice
+
+    def set_PAYGAS_gasprice(self, gas_price):
```

Comment:
I think that this is a good move (moving the value and the setters co `Computation`).  This function could use one extra validation check to enforce `depth == 0` within the setter itself.

Reply:
@pipermerriam 
Added the `depth == 0` validation check along with PAYGAS testing contract fixture and test suite.